### ISSUE TITLE
Tests for atom-shell startup

### ISSUE THREAD

**Issue by springmeyer** [mapbox/mapbox-studio-classic#1208]
We learned recently (#1166) that it is possible for all appveyor tests to pass even if the the windows app is completely unable to start. In the case of #1166 I think was happened was the server process started up not the shell (which left the server as a zombie).

To protect against releasing another broken build we need to:
- [ ] Dig into what happened in more detail in order to figure out where the missing test coverage is
- [ ] Add tests that cover the previous case (for tilemill we used to test app startup like [this](https://github.com/mapbox/tilemill-builder-osx/blob/master/build2.sh#L124-L139))
- [ ] Try reverting https://github.com/mapbox/mapbox-studio/commit/5a5e753154f808af802945c1c83cf1f0b994618a and ensure the tests catch this fail on windows
- [ ] Try creating testcases emulating an app that cannot startup on linux and ensure travis tests can catch this.

As far as finding a fix, so that https://github.com/mapbox/mapbox-studio/commit/5a5e753154f808af802945c1c83cf1f0b994618a can be safely reverted (and shell.log logging restored for windows) we are still empty handed. As far as I know the problem is still a mystery. My hunch is that we are being bitten by https://github.com/joyent/node/issues/3584. If this is the case then node v0.12.x should fix the problem. Or using a 0.10.x version with the blocking io fixes backported (refs https://github.com/mapbox/node-cpp11/issues/18).

/cc @camilleanne @yhahn @bsudekum

**Comment by yhahn** [comment]
Yes -- we currently don't have any tests around the atom shell startup process (wonder if we can do this headlessly or if that is its own can of worms...)

We'll have to dig into this to see what's possible -- in the meantime we should consider any changes to the atom shell portion of studio something someone must test manually on all 3 platforms.

**Comment by springmeyer** [comment]
@yhahn we used to test tilemill startup like [this](https://github.com/mapbox/tilemill-builder-osx/blob/master/build2.sh#L124-L139). Maybe a good place to start would be to add a cheap test like this to travis and mock ways that the shell might fail to start on unix.

**Comment by yhahn** [comment]
Sounds good. I bet we can monitor the app log as well to get a feel for if the app is starting (but hanging) or getting to other stages of startup.

**Comment by bsudekum** [comment]
@springmeyer I do like the look of this a lot: 

``` bash
killall node 2>/dev/null
killall tilemill-tile 2>/dev/null
killall tilemill-ui 2>/dev/null
```

When you say 'tests' are you talking about tests that run on every start or unit testing?

**Comment by springmeyer** [comment]
@bsudekum unit tests that run on travis/appveyor every commit.

**Comment by springmeyer** [comment]
@bsudekum - interested in working on this or have other ideas?
